contracts/base/ModuleManager.sol

```diff
@@ -139,10 +151,57 @@ contract ModuleManager is Storage, Receiver, IModuleManagerEventsAndErrors {
         ams.validators.init();
     }
 
+    /// @dev Implements Module Enable Mode flow.
+    /// @param module The address of the module to be installed.
+    /// @param packedData Data source to parse data required to perform Module Enable mode from.
+    /// @return userOpSignature the clean signature which can be further used for userOp validation
+    function _enableMode(address module, bytes calldata packedData) internal returns (bytes calldata userOpSignature) {   
+        uint256 moduleType;
+        bytes calldata moduleInitData;
+        bytes calldata enableModeSignature;
+        
+        (moduleType, moduleInitData, enableModeSignature, userOpSignature) = packedData.parseEnableModeData();  
+
+        _checkEnableModeSignature(
+            _getEnableModeDataHash(module, moduleInitData),
+            enableModeSignature
+        );
+        _installModule(moduleType, module, moduleInitData);
+    }
+
+    /// @notice Installs a new module to the smart account.
+    /// @param moduleTypeId The type identifier of the module being installed, which determines its role:
+    /// - 1 for Validator
+    /// - 2 for Executor
+    /// - 3 for Fallback
+    /// - 4 for Hook
```

Comment:
Would it be beneficial to include multitype here as well?

Reply:
done